src/assets/src/components/preferences.tsx

```diff
@@ -111,11 +187,49 @@ function PreferencesEditor(props: PreferencesEditorProps) {
         <div>
             <h1>View/Update Preferences</h1>
             {alertBlock}
-            <Form onSubmit={validateAndSubmit}>
-                <p>Enter a phone number in order to opt in to SMS notifications.</p>
-                <FormGroup controlId='phone' className="mb-3">
-                    <Form.Label>Phone Number</Form.Label>
-                    {phoneInput}
+            <h2>Contact Information</h2>
+            {phoneUpdateStatus === "" &&
+                <Form>
+                    <p>Enter a phone number in order to opt in to SMS notifications.</p>
+                    <FormGroup controlId='phone' className="mb-3">
+                        <Form.Label>Phone Number</Form.Label>
+                        {phoneInput}
+                    </FormGroup>
+
+                    {!sendingCode && changedPhoneNumber ? <Button variant="secondary" type="submit" disabled={props.disabled} onClick={getOneTimePassword}>Obtain a one-time phone verification code</Button>
+                        : sendingCode 
+                            ? <Button variant="secondary"><Spinner animation="border" size="sm" as="span" role="status" /> Sending...</Button>
+                            : null }
+                </Form>
+            }
+            {phoneUpdateStatus === "verify" &&
+                <Form>
+                    <p>Enter the verification code sent to {formattedPhoneNumberToSubmit} (<a onClick={() => setPhoneUpdateStatus("")}
+                        className="link-primary">edit</a>)</p>
+                    <FormGroup controlId='otp' className="mb-3">
+                        <Form.Control type="number" value={otpValue}
```

Comment:
User can only type in 4 digit in this input field, but the field width is much bigger than 4 digit. Could the width be adjusted accordingly? And it would be good to remove the "up" and "down" arrows on the right, too. 

Ideally, the OTP input fields shall be an array of one-digit input fields, like [an sample implementation here](https://medium.com/@tejas.shirnalkar/building-otp-and-password-inputs-with-validation-in-react-step-by-step-fda703742088) If you do not have time to adjust, please file an new issue for future enhancement.

Reply:
I initially was thinking of using [React-digit-input](https://www.npmjs.com/package/react-digit-input) but was facing errors. I will try to get it working or implement it manually.